### PR TITLE
[script] install latest CMake with pip

### PR DESCRIPTION
**Pull request by wgtdkp**
This PR installs latest CMake with python3-pip rather than building from source.

### CODE REVIEW

File: script/bootstrap.sh [hunk 1]
```diff
@@ -99,11 +72,14 @@ if [ $(uname) = "Linux" ]; then
...
                          ninja-build \
+                         python3-setuptools \
+                         python3-pip \
                          lcov -y
 
-    ## Install cmake 3.16.1
+    ## Install newest CMake
     match_version $(cmake --version | egrep -o '[0-9].*') ${MIN_CMAKE_VERSION} || {
-        install_cmake "3.16.1"
+        pip3 install -U pip
```
Comment (simonlingoogle) [inline script/bootstrap.sh]: Is it necessary to update pip ?

Comment (wgtdkp) [reply script/bootstrap.sh]: yes. otherwise, installing cmake will fail.

File: script/bootstrap.sh [hunk 2]
```diff
@@ -99,11 +72,14 @@ if [ $(uname) = "Linux" ]; then
...
+        pip3 install -U cmake
```
Comment (simonlingoogle) [inline script/bootstrap.sh]: If path is not setup correctly, users can still end up using wrong `cmake`. Should we check the `cmake` version after installation and warn the user loudly if version is not satisfied?

**Comment by wgtdkp** [comment]
Depending on https://github.com/openthread/ot-commissioner/pull/20 to get travis builds passed.

File: script/bootstrap.sh [hunk 3]
```diff
@@ -99,12 +72,19 @@ if [ $(uname) = "Linux" ]; then
...
+    ## Install newest CMake
+    match_version $(cmake --version | egrep -o '[0-9].*') ${MIN_CMAKE_VERSION} || {
+        pip3 install -U pip
+        pip3 install -U cmake
+    }
     match_version $(cmake --version | egrep -o '[0-9].*') ${MIN_CMAKE_VERSION} || {
-        install_cmake "3.16.1"
-        cmake --version
+        echo "error: cmake version($(cmake --version)) < ${MIN_CMAKE_VERSION}."
+        echo "did you forget to add '\$HOME/.local/bin' to beginning of your PATH?"
```
Comment (simonlingoogle) [inline script/bootstrap.sh]:
```suggestion
        echo "Did you forget to add '\$HOME/.local/bin' to beginning of your PATH?"
```

File: script/bootstrap.sh [hunk 4]
```diff
@@ -99,12 +72,19 @@ if [ $(uname) = "Linux" ]; then
...
                          cmake \
                          ninja-build \
+                         python-setuptools \
+                         python-pip \
                          lcov -y
 
-    ## Install cmake 3.16.1
+    ## Install newest CMake
+    match_version $(cmake --version | egrep -o '[0-9].*') ${MIN_CMAKE_VERSION} || {
+        pip install -U pip
```
Comment (wgtdkp) [inline script/bootstrap.sh]: The integration test uses OT Commissioner THCI which requires python2 by default. We here stick to python2 and see if we can upgrade to python3 in a future PR.

**Pull request merged**